scripts/confidence_ensembles/build_ensemble.py

```diff
@@ -70,7 +78,71 @@ class EnsembleConfig:
     # 100 is most likely enough, but setting higher default just in case
     max_training_samples: int = 1000
     # specify to provide dev data manifest for HP tuning
-    # dev_manifest: Optional[str] = None
+    dev_manifest: Optional[str] = None
+
+
+@dataclass
+class TuneConfidenceConfig:
+    # important parameter, so should always be tuned
+    exclude_blank: Tuple[bool] = (True, False)
+    # prod is pretty much always worse, so not including by default
+    aggregation: Tuple[str] = ("mean", "min", "max")
+    # not including max prob, as there is always an entropy-based metric
+    # that's better but otherwise including everything
+    confidence_type: Tuple[str] = (
+        "entropy_renui_exp",
+        "entropy_renui_lin",
+        "entropy_tsallis_exp",
+        "entropy_tsallis_lin",
+        "entropy_gibbs_lin",
+        "entropy_gibbs_exp",
+    )
+
+    # TODO: currently it's not possible to efficiently tune temperature, as we always
+    #    apply log-softmax in the decoder, so to try different values it will be required
+    #    to rerun the decoding, which is very slow. To support this for one-off experiments
```

Comment:
If you change temp, even if you had cached the decoder output without softmax, you'd still have to calculate the full decoding no ?

Reply:
Only if you use beam-search, which I never tried. But for greedy decoding, temperature does not change predictions, but can have a big effect on the confidence scores

Reply:
Interesting ok.